src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -60,7 +60,7 @@ def battery_inverter_mapping(batteries: Iterable[int]) -> dict[int, int]:
 
 # Formula output types class have no common interface
 # Print all possible types here.
-T = TypeVar("T", Sample[Percentage], Sample[Energy], PowerMetrics, Sample[Temperature])
+T = TypeVar("T", Sample[Percentage | None], Sample[Energy | None], PowerMetrics, Sample[Temperature | None])
```

Comment:
This change is not necessary, because `Sample` already has typing for `None` inside it.  Same in the return types below: https://github.com/frequenz-floss/frequenz-sdk-python/blob/3551cd3052efc9f60aa71b0acd2e3174c5c81a8a/src/frequenz/sdk/timeseries/_base_types.py#L18-L30